portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,40 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+
 /**
  * Adds a random greeting to the page.
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+function addGreeting() {
+  const greetingPromise = fetch("/data");
+  greetingPromise.then(handleGreeting);
+}
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+function handleGreeting(response) {
```

Comment:
nit: add JS function comment to all the new functions included in this PR. 

https://google.github.io/styleguide/jsguide.html#jsdoc-method-and-function-comments